### PR TITLE
Switch the HealthChecker to use the 'gardener.cloud/role' label key

### PR DESCRIPTION
**Pull request by ialidzhikov**
/kind cleanup

**What this PR does / why we need it**:
This PR:
- adapts the HealthChecker to use the `gardener.cloud/role` label key to select controlplane components (Deployments, StatefulSets, Etcds) for health check. This is follow-up on https://github.com/gardener/gardener/pull/3231.
- removes the `garden.sapcloud.io/role` label key from most of the controlplane components as it is no longer used for health checks.
- contains also some minor cleanups

**Which issue(s) this PR fixes**:
Part of #1649 

**Release note**:

```breaking operator
gardenlet's shoot-care-control is now using the `gardener.cloud/role` label key (until now it was `garden.sapcloud.io/role`) to perform health checks on controlplane components. Make sure you have first upgraded to at least Gardener v1.14 before you upgrade to this version of Gardener.
```

### CODE REVIEW

File: charts/seed-bootstrap/charts/loki/values.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ authEnabled: true
 annotations: {}
 
 labels:
-  # deprecated label only for the health check
+  # deprecated label
   garden.sapcloud.io/role: logging
```
Comment (rfranzke) [inline charts/seed-bootstrap/charts/loki/values.yaml:16]: Why is this not yet removed?

**Review by rfranzke — COMMENTED** [review]
looks great, just a small question

File: charts/seed-bootstrap/charts/loki/values.yaml (hunk 1 above)
Comment (ialidzhikov) [reply charts/seed-bootstrap/charts/loki/values.yaml:16]: Well, the `labels` are also used in the loki's Statefulset `spec.selector` which is immutable field. To workaround a similar issue with immutability of `spec.selector` for Deployment in https://github.com/gardener/gardener/pull/3264 I introduced a helper func that deletes the Deployment to allow new Deployment to be create right away with the desired spec. The same approach I think is not applicable for StatefulSets as they bring a PVC which we don't want to delete.

Comment (rfranzke) [reply charts/seed-bootstrap/charts/loki/values.yaml:16]: When a `StatefulSet` is deleted then the PVC remains in the system (at least not with our configuration), so I guess we can follow the same approach. We can do that separately, of course, so I'm going to approve the PR for now. Thanks!

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: charts/seed-bootstrap/charts/loki/values.yaml (hunk 1 above)
Comment (ialidzhikov) [reply charts/seed-bootstrap/charts/loki/values.yaml:16]: Okay, I will give a try. Thank you!

Comment (ialidzhikov) [reply charts/seed-bootstrap/charts/loki/values.yaml:16]: I now updated the PR. Thank you once again!

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/operation/botanist/logging.go [hunk 2]
```diff
@@ -74,5 +74,13 @@ func (b *Botanist) DeploySeedLogging(ctx context.Context) error {
 		}
 	}
 
+	// TODO (ialidzhikov): remove in a future version
+	stsKeys := []client.ObjectKey{
+		kutil.Key(b.Shoot.SeedNamespace, "loki"),
+	}
+	if err := common.DeleteStatefulSetsHavingDeprecatedRoleLabelKey(ctx, b.K8sSeedClient.Client(), stsKeys); err != nil {
```
Comment (timuthy) [inline pkg/operation/botanist/logging.go]: Aren't we required to wait for the STS to be gone before continuing?

Comment (ialidzhikov) [reply pkg/operation/botanist/logging.go]: I reverted this STS migration logic for now. I will tackle it in another PR as suggested initially.

**Review by rfranzke — APPROVED** [review]

**Review by timuthy — APPROVED** [review]

**Pull request merged**